### DISCUSSION TITLE
Flow typings for @babel/parser

### DISCUSSION THREAD

**Discussion by Kapelianovych** [babel/babel#16666] [Q&A] [answered]
Hello. The `7.8.4` version of `@babel/parser` had typings for FlowJS, but the latest version at the moment (`7.24.8`) does not (only *.d.ts* exists). Does anybody know when Flow typings were removed and where to? I checked but `flow-typed` does not contain them.

Update: apparently [npm shows no flow typings](https://www.npmjs.com/package/@babel/parser/v/7.8.4?activeTab=code) even for `7.8.4` version.

**Comment by JLHwung** [comment, marked as answer]
Babel parser has never published flow typings since Babel 7.

Before 7.19.0 (https://github.com/babel/babel/pull/14783), Babel parser was authored in flow but since then we have migrated the whole Babel monorepo to typescript.

The `flow-typed` is a third party repo not affiliated with us, currently we don't have plans to maintain flow types for Babel parser. If you would like a flow interface, you can start from our [typescript declarations](https://github.com/babel/babel/blob/main/packages/babel-parser/typings/babel-parser.source.d.ts) and open a PR to the `flow-typed` repo.

**Reply by Kapelianovych** [reply]
Thank you. I was observing a repository written with Flow which used `@babel/parser` programmatically and didn't see any other dependency for providing types, so I made a conclusion that `@babel/parser` provided Flow types by itself which was a mistake.
